clap_lex/src/lib.rs

```diff
@@ -466,3 +464,20 @@ fn split_nonutf8_once(b: &OsStr) -> (&str, Option<&OsStr>) {
         }
     }
 }
+
+fn looks_like_number(arg: &str) -> bool {
+    // strip a leading `-` in case this is a negative number
+    let arg = arg.strip_prefix("-").unwrap_or(arg);
+
+    // Return true if this looks like an integer or a float where it's all
+    // digits plus an optional single dot after some digits.
+    let mut seen_dot = false;
```

Comment:
What about the exponent?

Now, this also doesn't cover nan or inf but I would chalk that up to a bug in clap.

Reply:
Oh, missed the comment on edge cases :)

Reply:
I'm definitely happy to fill in more pieces here myself, but wanted to confirm that it's intentional to support the various float-related string forms before adding them in here as well. I'd naively assume, without having dug in further, that this was probably added to support `./foo.exe -1` with the argument `-1` instead of a flag `1`, but that's just a guess. In such a situation parsing `+1` may not be too important but parsing `-1e10` may be.

Reply:
Ok well as I say that I dug in more. Usage of `is_number` is always prefixed by `is_allow_negative_numbers_set`, so I could alternatively work backwards from there and try to get things configurable for LLVM to always const-propagate `false` there or something like that

Reply:
We could possibly specialize the function to `is_negative_number`.  Breaking changes in `clap_lex` aren't a big deal atm

Reply:
Just wanted to check on the idea of handling `e`

Reply:
Oops indeed, added in now (and updated tests too)